### PR TITLE
Add redux and fetching data from API

### PR DESCRIPTION
**Pull request by rana123bilal**
Done:
- [1] - Fetch Data with actions and passed data to components with redux.
- [2] -  All data fetches are implemented as actions and received from store by components
- [3] -  Sorting by release date and rating is implemented as redux actions
- [4] - Filtering by genre is implemented as redux actions

Not done:
nothing,

Final mark: 5

### CODE REVIEW

File: src/components/movie/Movies.css [hunk 1]
```diff
@@ -73,6 +73,18 @@ h2{
     font-style: normal;
     padding-left: 17px;
 }
```
Comment (Vladlan) [inline src/components/movie/Movies.css:75]: remove commented code

File: src/components/search/Search.js [hunk 2]
```diff
@@ -1,19 +1,41 @@
...
+    "family",
+    "mystery",
+    "science fiction",
+    "thriller",
+    "action",
+    "comedy",
+    "crime",
+  ];
 
   const searchButtonHandler = (event) => {
```
Comment (Vladlan) [inline src/components/search/Search.js:30]: searchButtonHandler -> searchMovies (or  doMoviesSearch or findMovies or smth like that)

File: src/index.js [hunk 3]
```diff
@@ -3,12 +3,14 @@ import ReactDOM from "react-dom/client";
 import { DataProvider } from "./context/data-context";
 import "./index.css";
 import App from "./App";
+import { Provider } from "react-redux";
+import { store } from "./store";
 
 const root = ReactDOM.createRoot(document.getElementById("root"));
 root.render(
-  <React.StrictMode>
```
Comment (Vladlan) [inline src/index.js:9]: Return back strict mode

File: src/redux/actions/movie-actions.js [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+    const response = await axios.get(`/movies?search=${search}&searchBy=title`);
+    dispatch({
+      type: SEARCH_LIST_BY_TITLE_SUCCESS,
+      payload: response.data.data,
+    });
+  } catch (error) {
+    dispatch({
+      type: SEARCH_LIST_BY_TITLE_FAIL,
+      payload:
+        error.message && error.response.data.message
```
Comment (Vladlan) [inline src/redux/actions/movie-actions.js:67]: Please check this logic and correct it `(error.message && error.response.data.message)`

File: src/redux/reducers/movie-reducer.js [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+      return {
+        loading: false,
+        error: true,
+      };
+    default:
+      return state;
+  }
+}
+
+export const movieDetailReducer = (state = intialMovieDetailState, action) => {
```
Comment (Vladlan) [inline src/redux/reducers/movie-reducer.js:113]: `, {type, payload}`

**Review by Vladlan — CHANGES_REQUESTED** [review]

File: src/redux/reducers/movie-reducer.js (hunk 5 above)
Comment (rana123bilal) [reply src/redux/reducers/movie-reducer.js:113]: Hey Vladislav,
could you please explain that what do you mean by {type, payload} here?

File: src/components/movie/Movies.css (hunk 1 above)
Comment (rana123bilal) [reply src/components/movie/Movies.css:75]: Commented code removed

File: src/components/search/Search.js (hunk 2 above)
Comment (rana123bilal) [reply src/components/search/Search.js]: changed the name of function to searchMovies

File: src/index.js (hunk 3 above)
Comment (rana123bilal) [reply src/index.js:8]: strict mode added

File: src/redux/actions/movie-actions.js (hunk 4 above)
Comment (rana123bilal) [reply src/redux/actions/movie-actions.js]: now error.message logic is working perfectly

**Review by Vladlan — APPROVED** [review]

**Pull request merged**